chia/daemon/keychain_proxy.py

```diff
@@ -89,13 +97,80 @@ def format_request(self, command: str, data: Dict[str, Any]) -> WsRpcMessage:
 
         return super().format_request(command, data)
 
+    async def _get(self, request: WsRpcMessage) -> WsRpcMessage:
+        """
+        Overrides DaemonProxy._get() to handle the connection state
+        """
+        try:
+            if not self.disconnect:  # if we are disconnected, and we send a request we should throw original error.
+                await asyncio.wait_for(self.connection_established.wait(), timeout=10)  # is 10 seconds too long?
+            else:
+                self.log.error("Attempting to send request to a keychain-proxy that has shut down.")
+            self.log.debug(f"Sending request to keychain: {request}")
```

Comment:
There's some sensitive data that can be logged here. We can log the command name but the request's `data` should not be logged.

Reply:
ah yeah very good point.